Ardagryd.js

```diff
@@ -597,61 +660,60 @@ export class Grid extends React.Component {
 
     constructor(props){
         super(props);
+
         this.state = {
             config: props.config ? props.config : {},
-            columns: props.columns ? props.columns : {},
+            filter: filterConfigFromProp(props.filter),
+            sort: sortConfigFromProp(props.sort),
             skip: 0
         };
         this.dispatch = this.dispatch.bind(this);
     }
 
     componentWillReceiveProps(nextProps){
-      this.setState({
-          config: nextProps.config ? nextProps.config : {},
-          columns: nextProps.columns ? nextProps.columns : {}
-      });
+        if (!(JSON.stringify(this.props.sort) === JSON.stringify(nextProps.sort))){
+            this.setState({sort: nextProps.sort});
+        }
+        if (!(JSON.stringify(this.props.filter) === JSON.stringify(nextProps.filter))){
+            this.setState({filter: nextProps.filter});
```

Comment:
It's probably more efficient to create a `newState` object with both changes combined and only call `setState` once. I'm not sure if it's advisable to only call `setState` if the object is non-empty though.